test/trace.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This file exports an interface that is identical to that of the Trace Agent,
+ * for testing purposes. The differences are that:
+ * - The Trace Writer singleton is mocked to make no network requests, writing
+ * traces to a local store instead.
+ * - When started, the Trace Agent is initialized with a samplingRate of zero by
+ * default (but this can be overridden).
+ * - Additional methods to query/delete spans written locally are exposed.
+ *
+ * Most tests should include this file instead of the main module root.
+ */
+
+// This is required for @google-cloud/common types.
+// tslint:disable-next-line:no-reference
+/// <reference path="../src/types.d.ts" />
+
+import * as common from '@google-cloud/common';
+import * as assert from 'assert';
+
+import * as trace from '../src';
+import {Config, PluginTypes} from '../src';
+import {SpanData} from '../src/span-data';
+import {Trace} from '../src/trace';
+import {TraceSpan} from '../src/trace-span';
+import {LabelObject, TraceWriter, traceWriter, TraceWriterConfig, TraceWriterSingletonConfig} from '../src/trace-writer';
+
+export {Config, PluginTypes};
+
+const traces = new Map<string, TraceSpan[]>();
+const allSpans: TraceSpan[] = [];
+
+class TestTraceWriter extends TraceWriter {
+  constructor(logger: common.Logger, config: TraceWriterConfig) {
+    super(logger, config);
```

Comment:
I'm not sure but doesn't it just work if this constructor definition is removed?

Reply:
Yep -- did not know this. Updated